workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/InviteMemberModal/InviteMemberModal.tsx

```diff
@@ -0,0 +1,309 @@
+      return { roleName: x, isChecked: false };
+    });
+    setCheckboxRoles(mapRoles);
+    getAccoutRoles();
+  }, []);
+
+  const jsonBody = () => {
+    return {
+      alternateName: email.replace('@', '-'),
+      emailAddress: email,
+      familyName: lastName,
+      givenName: firstName,
+    };
+  };
+
+  const getAccoutRoles = async () => {
+    const accountRoles = await fetch(
+      `/o/headless-admin-user/v1.0/accounts/${selectedAccount.id}/account-roles`,
+      {
+        headers: {
+          accept: 'application/json',
+          'x-csrf-token': Liferay.authToken,
+        },
+      }
+    );
+    if (accountRoles.ok) {
+      const data = await accountRoles.json();
+      setAccountRoles(data.items);
+    }
+  };
+
+  async function createNewUserIntoAccount() {
+    try {
+      const response = await fetch(
+        `/o/headless-admin-user/v1.0/accounts/${selectedAccount.id}/user-accounts`,
+        {
+          headers: {
+            accept: 'application/json',
+            'Content-Type': 'application/json',
+            'x-csrf-token': Liferay.authToken,
+          },
+          method: 'POST',
+          body: JSON.stringify(jsonBody()),
+        }
+      );
+    } catch (error) {
+      <ClayAlert.ToastContainer>
+        <ClayAlert
+          autoClose={5000}
+          displayType="danger"
+          title="error"
+        ></ClayAlert>
+      </ClayAlert.ToastContainer>;
+    }
+  }
+
+  async function addExistentUserIntoAccount() {
+    try {
+      const response = await fetch(
+        `/o/headless-admin-user/v1.0/accounts/${selectedAccount.id}/user-accounts/by-email-address/${email}`,
+        {
+          headers: {
+            accept: 'application/json',
+            'x-csrf-token': Liferay.authToken,
+          },
+          method: 'POST',
+          body: JSON.stringify(jsonBody),
+        }
+      );
+    } catch (error) {
+      <ClayAlert.ToastContainer>
+        <ClayAlert
+          autoClose={5000}
+          displayType="danger"
+          title="error"
+        ></ClayAlert>
+      </ClayAlert.ToastContainer>;
+    }
+  }
+
+  const getUserByEmail = async (userEmail: String) => {
+    try {
+      const responseFilteredUserList = await fetch(
+        `/o/headless-admin-user/v1.0/user-accounts?filter=emailAddress eq '${userEmail}'`,
+        {
+          headers: {
+            accept: 'application/json',
+            'x-csrf-token': Liferay.authToken,
+          },
+        }
+      );
+
+      if (responseFilteredUserList.ok) {
+        const data = await responseFilteredUserList.json();
+        if (data.items.length > 0) {
+          return data.items[0];
+        }
+      }
+    } catch (error) {
+      <ClayAlert.ToastContainer>
+        <ClayAlert
+          autoClose={5000}
+          displayType="danger"
+          title="error"
+        ></ClayAlert>
+      </ClayAlert.ToastContainer>;
+    }
+  };
+
+  const addAccountRolesToUser = async (user: any) => {
```

Comment:
Please change to the correct type here.
```suggestion
  const addAccountRolesToUser = async (user: ???) => {
```